contrib/systemd/gitea.service

```
@@ -57,6 +57,12 @@ WorkingDirectory=/var/lib/gitea/
 ExecStart=/usr/local/bin/gitea web --config /etc/gitea/app.ini
 Restart=always
 Environment=USER=git HOME=/home/git GITEA_WORK_DIR=/var/lib/gitea
+# If you install Git to directory prefix other than default PATH (which happens
+# for example if you install other versions of Git side-to-side with
+# distribution version), uncomment below line and add that prefix to PATH
+# Don't forget to place git-lfs binary on the PATH below if you want to enable
+# Git LFS support
+#Environment=PATH=/path/to/git/bin:/bin:/sbin:/usr/bin:/usr/sbin
```

```suggestion
#Environment=PATH=/path/to/git/dir:${PATH}
```

Not totally sure if this will work but documentation indicates it does. Certainly better than hardcoding a path.

---

@silverwind I think `/bin` directory of `/path/to/git` prefix must be added to PATH, because there is where `git` binary is located there.

---

@silverwind Hmmm... applying your suggestion on real service (on my instance) returned 500 when trying to create new repo. It seems
like the error caused by `bash` was not found on PATH:

```
routers/repo/repo.go:182:handleCreateError() [E] CreatePost: initRepository: initRepoCommit: git push: exit status 1 - remote: /usr/bin/env: 'bash': No such file or directory
        To /srv/git/repo/someone/test.git
         ! [remote rejected] master -> master (pre-receive hook declined)
        error: failed to push some refs to '/srv/git/repo/someone/test.git'
```

I guessed that on that case, systemd expands `PATH=/path/to/git/bin:${PATH}` to just `PATH=/path/to/git/bin`, thus missing default PATHs.

So one need to specify PATH with Git prefix (`/path/to/git/bin`) **AND** default PATHs (which in most cases are `/bin:/sbin:/usr/bin:/usr/sbin`).

---

> because that is where git binary is located there. Note that bin is significant.

Is `bin` a folder when you compile git from source? It seemed confusing the me because one could interprest `bin` as the actual binary, that's why I was suggesting `dir` to make it clear it's a directory.

> Hardcoding PATH is required for systemd service to work properly.

Yes, turns out variable expansion is not possible inside `Environment=`, only inside `ExecStart` as per [this](https://www.freedesktop.org/software/systemd/man/systemd.exec.html).

---

@silverwind I will explain about prefix directory.

On packages which use `./configure && make && make install` idiom (*CMMI*, `git` included), you can specify `prefix` (base directory which the packages will be installed there) by `--prefix=/some/where`. So for git, that will be:
```bash
cd git-<version>
./configure --prefix=/path/to/git-prefix
make && make install
```

To use Git that installed on such prefix, you would call `git` either by specifying full prefix to the binary:
```bash
/path/to/git-prefix/bin/git version
```
or by setting `PATH` first, then just call the binary name:
```bash
export PATH=/path/to/git-prefix/bin:$PATH
git version
```

Note that CMMI packages install their binaries to `bin/` directory of the prefix (for example `/some/where/bin` contains binary named `foobar`, and the path to `foobar` is `/some/where/bin/foobar`).

When you suggest using `dir/` (as in `/path/to/git/dir`) you implied that you just specify the prefix without `bin/` directory
(which in that case would be `/path/to/git/dir/bin`).

---

Right, I never compiled git so this seems right for that use case.

---

@silverwind ok